app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx

```diff
@@ -15,146 +15,73 @@
-      if (connectorType.name === 'S3') {
-        return {
-          ...connectorType,
-          SVG: S3,
-        };
-      } else if (connectorType.name === 'Database') {
-        return {
-          ...connectorType,
-          SVG: Database,
-        };
-      } else if (connectorType.name === 'BigQuery') {
-        return {
-          ...connectorType,
-          SVG: BigQuery,
-        };
-      } else if (connectorType.name === 'GCS') {
-        return {
-          ...connectorType,
-          SVG: GCS,
-        };
-      } else if (connectorType.name === 'Spanner') {
-        return {
-          ...connectorType,
-          SVG: Spanner,
-        };
-      } else if (connectorType.name === 'Kafka') {
-        return {
-          ...connectorType,
-          SVG: Kafka,
-        };
-      } else if (connectorType.name === 'SQL Server') {
-        return {
-          ...connectorType,
-          SVG: SQLServer,
-        };
-      } else if (connectorType.name === 'MySQL') {
-        return {
-          ...connectorType,
-          SVG: MySQL,
-        };
-      } else if (connectorType.name === 'Oracle') {
-        return {
-          ...connectorType,
-          SVG: Oracle,
-        };
-      } else if (connectorType.name === 'PostgreSQL') {
-        return {
-          ...connectorType,
-          SVG: PostgreSQL,
-        };
-      } else if (connectorType.name === 'File') {
-        return {
-          ...connectorType,
-          SVG: ImportDatasetIcon,
-        };
-      } else if (connectorType.name === 'CloudSQLMySQL') {
-        return {
-          ...connectorType,
-          SVG: CloudSQLMySQL,
-        };
-      } else if (connectorType.name === 'CloudSQLPostgreSQL') {
-        return {
-          ...connectorType,
-          SVG: CloudSQLPostGreSQL,
-        };
-      } else {
-        return {
-          ...connectorType,
-          SVG: BigQuery,
-        };
-      }
-    });
-
-    fetchedConnectorTypesFromAPI.unshift({
-      name: 'Imported Datasets',
-      type: 'default',
-      category: 'default',
-      description: 'All Connections from the List',
-      artifact: {
-        name: 'allConnections',
-        version: 'local',
-        scope: 'local',
-      },
 
-      SVG: ImportDatasetIcon,
-    });
+  const classes = useStyles();
+  const connectorTypes: IConnectorArray[] = connectorsData.connectorTypes;
 
-    setConnectorTypes({
-      fetchedConnectorTypes: fetchedConnectorTypesFromAPI,
-    });
+  const updateState = (updatedState: { connectorTypes: IConnectorArray[] }) => {
+    if (
+      updatedState.hasOwnProperty('connectorTypes') &&
+      Array.isArray(updatedState.connectorTypes) &&
+      updatedState.connectorTypes.length > 5
+    ) {
+      toggleViewAllLink(true);
+    } else {
+      toggleViewAllLink(false);
+    }
+    setConnectorsData(updatedState);
   };
 
   useEffect(() => {
-    getConnectorTypesNames();
+    getWidgetData(updateState);
```

Comment:
so this is only run one time?

Reply:
whenever the connectorsWithIcons data in store get changed. This will get re-run.